component/spi/storage/storage.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package storage
+
+import (
+	"errors"
+)
+
+// MultiError represents the errors that occurred during a bulk operation.
+type MultiError interface {
+	error
+	Errors() []error // Errors returns the error objects for all operations.
+}
+
+// ErrStoreNotFound is returned when a store is not found.
+var ErrStoreNotFound = errors.New("store not found")
+
+// ErrDataNotFound is returned when data is not found.
+var ErrDataNotFound = errors.New("data not found")
+
+// StoreConfiguration represents the configuration of a store.
+type StoreConfiguration struct {
+	TagNames []string // A list of Tag names that key + value pairs in this store can be associated with.
+}
+
+// QueryOptions represents various options for Query calls in a store.
+type QueryOptions struct {
+	// PageSize sets the page size used by the Store.Query method.
+	PageSize int
+}
+
+// QueryOption represents an option for a Query call in a store.
+type QueryOption func(opts *QueryOptions)
```

Comment:
Note: initially I was going to create a single spi package.

But currently this file is setup to be namespaced under a storage package so I left it for now.